apps/notifications/bin/prepArtifactForCI.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+set -x
+
+function get_hash {
+	# If the stylesheet in the HTML file is pointing at "build.min.css?foobar123",
+	# this will just return the "foobar123" portion of the file. This
+	# is a source of randomness which needs to be eliminated.
+	echo `sed -nE 's~.*<link rel="stylesheet" href="build.min.css\?([a-zA-Z0-9]+)">.*~\1~p' ${'$'}1`
+}
+
+# Copy the README file.
+cp ./README.md dist/
+
+new_hash=`get_hash dist/index.html`
+old_hash=`get_hash release-archive/index.html`
+
+# All scripts and styles use the same "hash" version, so replace any
+# instances of the hash in the *old* files with the newest version.
+sed -i "s~${'$'}old_hash~${'$'}new_hash~g" release-archive/index.html release-archive/rtl.html
```

Comment:
the weird dollar sign stuff isn't needed outside of kotlin files

Reply:
> the weird dollar sign stuff isn't needed outside of kotlin files

Gotcha, I am not exactly sure how this is used. 

Would replacing `${'$'}old_hash` with `$old_hash` work, for example?

Reply:
yes exactly!

The reason it's needed is because in a Kotlin string, the dollar sign is part of string interpolation:

```
val foo = "hello"
println( "$foo world" ) // prints "hello world"
```

But dollar signs are also how you write variables in Bash, so if you copy/paste a bash script into a kotlin string, then it breaks -- because Kotlin treats all the dollar signs as Kotlin variables. So you have to escape it for it to work in bash. And `${"$"}` is literally just Kotlin string interpolation that resolves to the dollar sign in the actual string:


```
println( "${'$'}100, please" ) // prints "$100, please"
```

You can use `\$`, but this doesn't work in multi-line strings like we use in writing bash scripts.